src/methylbert/trainer.py

```diff
@@ -578,18 +586,24 @@ def _iteration(self, steps, data_loader):
 
                         f_perform.write("\t".join([str(self.step), str(global_step_loss), str(train_ctype_acc),  str(self.optim.param_groups[0]["lr"])])+"\n")
 
-                    self.step += 1
-                    duration=0
-                    global_step_loss = 0
-
+                    steps_progress_bar.set_postfix(eval_loss=eval_loss)
                     # Reset prediction result
                     del train_prediction_res
                     train_prediction_res =  {"dmr_label":[], "pred_ctype_label":[], "ctype_label":[]}
-             
+
+                self.step += 1
+                duration=0
+                global_step_loss = 0
```

Comment:
I am a bit confused here. Wouldn't the global_step_loss need to be reset when the gradient accumulation step is achieved? I think this should be under "`if (local_step+1) % self._config.gradient_accumulation_steps == 0`". Maybe I should add one more line to divide `global_step_loss` with `gradient_accumulation_step` before recording the performance in `f_perform` file.

Reply:
You are right, I've only worked with gradient accumulation of 1, so I didn't notice that! It's up to you, don't find it necessary to add that division 🤔